File: python/6_kyu/convert_to_camel.py
# https://www.codewars.com/kata/517abf86da9663f1d2000003

def to_camel_case(text):
    output = []
    capitalize = False
    for chr in text:
        if chr == "-" or chr == "_":
            capitalize = True
        else:
            output.append(chr.upper() if capitalize else chr)
            capitalize = False
    return "".join(output)

File: CodeWars/javascript/6_kyu/convert_to_camel.js
// https://www.codewars.com/kata/517abf86da9663f1d2000003

function toCamelCase(str) {
  const output = [];
  let capitalize = false;
  for (let chr of str) {
    if (chr == "-" || chr == "_") capitalize = true;
    else {
      output.push(capitalize ? chr.toUpperCase() : chr);
      capitalize = false;
    }
  }
  return output.join("");
}
